File: app_codebase/dart_bkend_base_02/bin/modules/ops/ops_errors.dart
/// Ops module errors (drain / maintenance).
library;

import '../../core/errors/contracts/register_module_error_contract.dart';
import '../../core/errors/error_spec.dart';

/// HTTP/service drain rejection.
const drainModeError = ErrorSpec(
  'ops/drain_mode',
  'Server is in drain / maintenance mode',
  503,
  fatalWs: true,
);

/// WS close/reject code (Dutch-compatible name).
const serverMaintenance = ErrorSpec(
  'server_maintenance',
  'Server is in drain / maintenance mode',
  503,
  fatalWs: true,
);

void registerOpsErrors(ModuleErrorRegistrar registrar) {
  registrar.registerModule('ops', [drainModeError]);
}

File: app_codebase/python_base_05/bin/modules/ops/ops_errors.py
"""Ops module error codes."""

from __future__ import annotations

from core.errors.contracts.register_module_error_contract import ModuleErrorRegistrar
from core.errors.error_spec import ErrorSpec

DRAIN_MODE = ErrorSpec(
    "ops/drain_mode",
    "Server is in drain / maintenance mode",
    http_status=503,
)


def register_ops_errors(registrar: ModuleErrorRegistrar) -> None:
    registrar.register_module("ops", [DRAIN_MODE])
